### PR TITLE
Issue 46744: Row selectors in QueryWebPart with study schema

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
Currently, it is not possible to use LabKey.QueryWebPart with a study dataset and specify that you do not want selectors to appear. This fix resolves that issue while not appearing to affect default dataset row selection behavior.

#### Related Pull Requests
* N/A

#### Changes
* Remove code block setting selector override

### CODE REVIEW

File: study/src/org/labkey/study/query/DatasetQueryView.java [hunk 1]
```diff
@@ -414,12 +414,6 @@ protected void populateButtonBar(DataView view, ButtonBar bar)
         List<String> recordSelectorColumns = view.getDataRegion().getRecordSelectorValueColumns();
         bar.add(createExportButton(recordSelectorColumns));
 
-        // Duplicates logic from super.populateButtonBar(), but we need selectors
```
Comment (RosalineP) [inline study/src/org/labkey/study/query/DatasetQueryView.java:417]: Note that doing `(((recordSelectorColumns != null && !recordSelectorColumns.isEmpty()) || (getTable() != null && !getTable().getPkColumns().isEmpty()) && showRecordSelectors()))` is an alternative. Currently don't see an effect to getting rid of this code block, though.

**Comment by RosalineP** [comment]
Looks like some of the TC builds got removed partway through green runs—will merge up and re-trigger soon

**Review by labkey-nicka — COMMENTED** [review]
As discussed in person I think we should try to align the behavior of `DatasetQueryView` as much with `QueryView` as possible to ensure respecting view settings (i.e. QueryWebPart settings). There are a couple options as I see it:

1. Replicate the behavior from `QueryView` where the selection boxes are forced to display regardless of view settings when the export panels are available: 
```java
if (showExportButtons())
{
    List<String> recordSelectorColumns = view.getDataRegion().getRecordSelectorValueColumns();

    PanelButton exportButton = createExportButton(recordSelectorColumns);
    if (exportButton.hasSubPanels())
    {
        if ((recordSelectorColumns != null && !recordSelectorColumns.isEmpty()) || (getTable() != null && !getTable().getPkColumns().isEmpty()))
        {
            bar.setAlwaysShowRecordSelectors(true);
        }
        bar.add(exportButton);
    }
}
```
2. Fully remove the `recordSelectorColumns` lines like you have done but also remove the corollary logic in `QueryView` for respecting this when part of export. It seems odd to me that exporting would dictate the presence of selection checkboxes over/above a user supplied specific configuration asking not to display checkboxes. For this we'd want to verify the scenario in [Issue 24530](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=24530) to ensure that the checkboxes display as desired for audit log grids.

An additional discrepancy to note is that `DatasetQueryView` is not respecting `showExportButton()` either. Either way we go here I think we should probably fix that as well.

File: study/src/org/labkey/study/query/DatasetQueryView.java (hunk 1 above)
Comment (labkey-nicka) [reply study/src/org/labkey/study/query/DatasetQueryView.java:417]: As discussed in person I think we should try to align the behavior of `DatasetQueryView` as much with `QueryView` as possible to ensure respecting view settings (i.e. QueryWebPart settings). There are a couple options as I see it:

1. Replicate the behavior from `QueryView` where the selection boxes are forced to display regardless of view settings when the export panels are available: 
```java
if (showExportButtons())
{
    List<String> recordSelectorColumns = view.getDataRegion().getRecordSelectorValueColumns();

    PanelButton exportButton = createExportButton(recordSelectorColumns);
    if (exportButton.hasSubPanels())
    {
        if ((recordSelectorColumns != null && !recordSelectorColumns.isEmpty()) || (getTable() != null && !getTable().getPkColumns().isEmpty()))
        {
            bar.setAlwaysShowRecordSelectors(true);
        }
        bar.add(exportButton);
    }
}
```
2. Fully remove the `recordSelectorColumns` lines like you have done but also remove the corollary logic in `QueryView` for respecting this when part of export. It seems odd to me that exporting would dictate the presence of selection checkboxes over/above a user supplied specific configuration asking not to display checkboxes. For this we'd want to verify the scenario in [Issue 24530](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=24530) to ensure that the checkboxes display as desired for audit log grids.

An additional discrepancy to note is that `DatasetQueryView` is not respecting `showExportButton()` either. Either way we go here I think we should probably fix that as well.

Comment (RosalineP) [reply study/src/org/labkey/study/query/DatasetQueryView.java:417]: (After discussing in-person we opted for option 1)

**Review by labkey-nicka — APPROVED** [review]

**Pull request merged**